src/lib/util.h

```diff
@@ -65,6 +65,11 @@ char *rstrip(char *a, size_t len, ssize_t *newlen);
  */
 char *string_printf(const char *fmt, ...) __attribute__((format(printf, 1, 2)));
 
+/**
+ * Similar to string_prntf but append the formated string on the original string.
+ */
+char *string_append(char **original, const char *fmt, ...) __attribute__((format(printf, 2, 3)));
```

Comment:
Remove this function and use `string_printf` above. It does not make much sense since any you will be re-allocating most of the time anyways.

Reply:
doing it